packages/contentstack-audit/src/modules/entries.ts

```diff
@@ -770,6 +838,55 @@ export default class Entries {
     }
     return entry;
   }
+
+  validateMandatoryFields(tree: Record<string, unknown>[], fieldStructure: any, entry: any) {
+    const { display_name, multiple, data_type, mandatory, field_metadata, uid } = fieldStructure;
+
+    if (mandatory) {
+      if (data_type === 'json' && field_metadata.allow_json_rte) {
+        const isval = multiple
+          ? entry[uid]?.[0]?.children?.[0]?.children?.[0]?.text === ''
+          : entry[uid]?.children?.[0]?.children?.[0]?.text === '';
+
+        if (isval) {
+          return [
+            {
+              uid: this.currentUid,
+              name: this.currentTitle,
+              display_name,
+              missingFieldUid: uid,
+              tree,
+              treeStr: tree
+                .filter(({ name }) => name)
+                .map(({ name }) => name)
+                .join(' ➜ '),
+            },
+          ];
+        }
+      } else {
+        if (!entry[uid] || (multiple ? !entry[uid].length : entry[uid] === '')) {
+          return [
+            {
+              uid: this.currentUid,
+              name: this.currentTitle,
+              display_name,
+              missingFieldUid: uid,
```

Comment:
@cs-raj , we can make this code more readable. please check below suggested code and if it make sense to you then please incorporate it.
1. isJsonRTE and fieldValue outside of the conditional checks to avoid repeating check.
2. Avoid nested if statements

  if (!mandatory) {
    return [];
  }
  
  const isJsonRTE = data_type === 'json' && field_metadata.allow_json_rte;
  const fieldValue = entry[uid];
  const isValueEmpty = multiple
          ? entry[uid]?.[0]?.children?.[0]?.children?.[0]?.text === ''
          : entry[uid]?.children?.[0]?.children?.[0]?.text === '';
          
      if(isJsonRTE & isvalEmpty){
      //first case
      }
       if (!fieldValue || (multiple ? !fieldValue.length : fieldValue === '')) {
       //2nd case
       }
   return [];

Reply:
Added